test/FunctionalTests/Server/CompressionTests.cs

```diff
@@ -430,11 +430,12 @@ public async Task SendCompressedMessageWithoutEncodingHeader_ServerErrorResponse
             AssertHasLogRpcConnectionError(StatusCode.Internal, "Request did not include grpc-encoding value with compressed message.");
         }
 
-        [TestCase("gzip", true)]
+        [TestCase("gzip", "gzip", true)]
+        [TestCase("gzip", "identity, gzip", true)]
 #if NET6_0_OR_GREATER
-        [TestCase("deflate", false)]
+        [TestCase("deflate", "deflate", false)]
```

Comment:
```suggestion
        [TestCase("deflate ", "deflate", false)]
```

Should this test case include trailing whitespace as well just to cover the scenario where a single algo is used?